pkg/asset/ignition/ignition_v2.go

```diff
@@ -0,0 +1,356 @@
+				},
+			},
+		},
+	}
+}
+
+// InjectInstallInfo adds information about the installer and its invoker as a
+// ConfigMap to the provided bootstrap Ignition config.
+func InjectInstallInfo(bootstrap []byte) (string, error) {
+	config := &igntypes2.Config{}
+	if err := json.Unmarshal(bootstrap, &config); err != nil {
+		return "", errors.Wrap(err, "failed to unmarshal bootstrap Ignition config")
+	}
+
+	cm, err := openshiftinstall.CreateInstallConfigMap("openshift-install")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate openshift-install config")
+	}
+
+	config.Storage.Files = append(config.Storage.Files, FileFromString("/opt/openshift/manifests/openshift-install.yaml", "root", 0644, cm))
+
+	ign, err := json.Marshal(config)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to marshal bootstrap Ignition config")
+	}
+
+	return string(ign), nil
+}
+
+// GenerateMinimalConfig returns a minimal ignition v2 config
+func GenerateMinimalConfig() *Config {
+	return &Config{
+		Ignition: igntypes2.Ignition{
+			Version: igntypes2.MaxVersion.String(),
+		},
+	}
+}
+
+// AddSSHKey returns a minimal ignition v2 config
+func (c *Config) AddSSHKey(sshKey string) {
+	c.Passwd.Users = append(
+		c.Passwd.Users,
+		igntypes2.PasswdUser{Name: "core", SSHAuthorizedKeys: []igntypes2.SSHAuthorizedKey{igntypes2.SSHAuthorizedKey(sshKey)}},
+	)
+}
+
+// AddSystemdUnit appends contents in Ignition config
+func (c *Config) AddSystemdUnit(name string, contents string, enabled bool) {
+	unit := igntypes2.Unit{
+		Name:     name,
+		Contents: contents,
+	}
+	if enabled {
+		unit.Enabled = util.BoolToPtr(true)
+	}
+	c.Systemd.Units = append(c.Systemd.Units, unit)
+}
+
+// AddSystemdDropins appends systemd dropins in the config
+func (c *Config) AddSystemdDropins(name string, children []Dropin, enabled bool) {
+	dropins := []igntypes2.SystemdDropin{}
+	for _, childInfo := range children {
+
+		dropins = append(dropins, igntypes2.SystemdDropin{
+			Name:     childInfo.Name,
+			Contents: childInfo.Contents,
+		})
+	}
+	unit := igntypes2.Unit{
+		Name:    name,
+		Dropins: dropins,
+	}
+	if enabled {
+		unit.Enabled = util.BoolToPtr(true)
+	}
+	c.Systemd.Units = append(c.Systemd.Units, unit)
+}
+
+// ReplaceOrAppend is a function which ensures duplicate files are not added in the file list
+func (c *Config) ReplaceOrAppend(file igntypes2.File) {
+	for i, f := range c.Storage.Files {
+		if f.Node.Path == file.Node.Path {
+			c.Storage.Files[i] = file
+			return
+		}
+	}
+	c.Storage.Files = append(c.Storage.Files, file)
+}
+
+// To allow Ignition to download its config on the bootstrap machine from a location secured by a
+// self-signed certificate, we have to provide it a valid custom ca bundle.
+// To do so we generate a small ignition config that contains just Security section with the bundle
+// and later append it to the main ignition config.
+// We can't do it directly in Terraform, because Ignition provider suppors only 2.1 version, but
```

Comment:
I opened a PR a month ago to support Spec 2.4 in the provider: https://github.com/terraform-providers/terraform-provider-ignition/pull/67

Reply:
this may not be needed after all. If you don't have to use 2.4 everywhere, I'd say let's not in order to keep changes to a mininum.